Automattic/wp-calypso
# Update imports/exports to use destructuring properly

**summary**
There many files that need to be modified in order to [enable scope hoisting and tree shaking](https://github.com/Automattic/wp-calypso/pull/16057#issuecomment-335047339) and disable the cjs module transform. They need to be modified becaus we have been using es6 modules more loosely than actually [allowed by the spec](https://github.com/Automattic/wp-calypso/pull/16057#issuecomment-335047339). 

The nice thing about stacking most of the fixes into one big PR is that we can test it all by turning off cjs transform and having webpack guarantee that all of our imports are valid

---

This PR fixes all of the issues marked here:  https://github.com/Automattic/wp-calypso/pull/16057#issuecomment-335047339 and is a blocker for disabling commonjs transpilation.

To Review
-----
I recommend viewing the version of the diff [without whitespace](https://github.com/Automattic/wp-calypso/pull/18838/files?w=1).  90% of the changes are just the addition of extra exports.

To Test 
------
first test _without_ cjs transpilation (where we eventually want to get).  the point of this is to ensure that we haven't broken/missed any imports/exports and to make sure everything is still connected together properly.  If we messed up an import, webpack would tell us.  You can see the error reporting behavior locally by messing up an import and you'll see it in massive yellow font.
1. `npm start`.  
2. none of the files should still have **build** errors (runtime errors shall abound)

Then also test _with_ cjs transpilation.  This should work 100%
1. `git checkout update/exports-imports-es6-with-cjs`
2. npm start
